packages/block-editor/src/hooks/layout.js

```diff
@@ -212,9 +255,13 @@ export const withLayoutStyles = createHigherOrderComponent(
 		const usedLayout = layout?.inherit
 			? defaultThemeLayout
 			: layout || defaultBlockLayout || {};
-		const className = classnames( props?.className, {
-			[ `wp-container-${ id }` ]: shouldRenderLayoutStyles,
-		} );
+		const className = classnames(
+			props?.className,
+			{
+				[ `wp-container-${ id }` ]: shouldRenderLayoutStyles,
+			},
+			getLayoutClasses( attributes )
```

Comment:
This might be slightly nit-picky, but in the PHP implementation, the classnames will only be added if the block has opted-in to the layout support. Theoretically in this JS version, it looks like if the attributes have a `layout` property (with the appropriate sub-properties), it'll output the classnames even if it hasn't opted in to the layout support. I can only imagine this happening in an unfortunate edge case of a custom block in a plugin that happens to use a `layout` attribute.

Do you think it's worth adding a `shouldRenderLayoutStyles` check before calling `getLayoutClasses`?

Reply:
Good catch, thanks, will sort that out in the morning